lib/vnet/daemon/client_darwin.m

```diff
@@ -97,3 +72,90 @@ void OpenSystemSettingsLoginItems(void) {
     [SMAppService openSystemSettingsLoginItems];
   }
 }
+
+@interface VNEDaemonClient ()
+
+@property(nonatomic, strong, readwrite) NSXPCConnection *connection;
+@property(nonatomic, strong, readonly) NSString *bundlePath;
+
+@end
+
+@implementation VNEDaemonClient
+
+- (id)initWithBundlePath:(NSString *)bundlePath {
+  self = [super init];
+  if (self) {
+    _bundlePath = bundlePath;
+  }
+  return self;
+}
+
+- (NSXPCConnection *)connection {
+  // Create the XPC Connection on demand.
+  if (_connection == nil) {
+    _connection = [[NSXPCConnection alloc] initWithMachServiceName:DaemonLabel(_bundlePath)
+                                                           options:NSXPCConnectionPrivileged];
+    _connection.remoteObjectInterface =
+        [NSXPCInterface interfaceWithProtocol:@protocol(VNEDaemonProtocol)];
+    _connection.invalidationHandler = ^{
+      self->_connection = nil;
+    };
+
+    // New connections always start in a suspended state.
+    [_connection resume];
+  }
+  return _connection;
+}
+
+- (void)startVnet:(VnetConfig *)vnetConfig completion:(void (^)(NSError *))completion {
+  // This way of calling the XPC proxy ensures either the error handler or
+  // the reply block gets called.
+  // https://forums.developer.apple.com/forums/thread/713429
+  id proxy = [self.connection remoteObjectProxyWithErrorHandler:^(NSError *error) {
+    completion(error);
+  }];
+
+  [(id<VNEDaemonProtocol>)proxy startVnet:vnetConfig
+                               completion:^(void) {
+                                 completion(nil);
+                               }];
+}
+
+- (void)invalidate {
+  if (_connection) {
+    [_connection invalidate];
+  }
+}
+
+@end
+
+static VNEDaemonClient *daemonClient = NULL;
+
+void StartVnet(StartVnetRequest *request, StartVnetResult *outResult) {
+  if (!daemonClient) {
+    daemonClient = [[VNEDaemonClient alloc] initWithBundlePath:@(request->bundle_path)];
+  }
+
+  dispatch_semaphore_t sema = dispatch_semaphore_create(0);
```

Comment:
Please add "#include <dispatch/dispatch.h>" for dispatch_semaphore_t.

- https://google.github.io/styleguide/cppguide.html#Include_What_You_Use